.github/workflows/verible-ci.yml

```diff
@@ -4,99 +4,140 @@ on:
-          run:
-            ./.github/workflows/install.sh
-
-        - name: Main script
-          run: ./.github/workflows/script.sh
-
-        - name: Upload binary
-          uses: actions/upload-artifact@v2
-          with:
-            path: /tmp/releases/verible-*.tar.gz
-
-        - name: After success
-          id: after_success
-          if: ${{ success() }}
-          run: ./.github/workflows/success.sh
-
-        - name: Deployment
-          if: ${{ github.ref == 'refs/heads/master' && github.event_name != 'pull_request' && success() }}
-          uses: svenstaro/upload-release-action@v2
-          with:
-            repo_token: ${{ secrets.GITHUB_TOKEN }}
-            file: /tmp/releases/verible-*.tar.gz
-            tag: ${{ steps.after_success.outputs.TAG }}
-            overwrite: true
-            file_glob: true
+
+
+  Check:
+    runs-on: ubuntu-16.04
+    strategy:
+      fail-fast: false
+      matrix:
+        mode:
+          - test
+          - compile
+          - clean
+    env:
+      MODE: ${{ matrix.mode}}
+    name: ${{ matrix.mode}}
+    steps:
+
+    - name: Checkout code
+      uses: actions/checkout@v2
+      with:
+       # Download complete repository + tags
+       fetch-depth: 0
+
+    - name: Mount bazel cache
+      uses: actions/cache@v1
+      if: matrix.mode == 'clean'
+      with:
+        path: "/home/runner/.cache/bazel"
+        key: bazel-${{ env.MODE }}
+
+    - name: Before Install
+      run: |
+        set -x
+        source ./.github/settings.sh
+        ./.github/workflows/set-compiler.sh 9
+        ./.github/workflows/install-bazel.sh
+
+    - name: Main script
+      run: ./.github/workflows/script.sh
+
+    - name: Set up things for GitHub Pages deployment
+      if: matrix.mode == 'compile'
+      run: ./.github/workflows/github-pages-setup.sh
+
+    - name: Deployment
+      if: github.ref == 'refs/heads/master' && github.event_name != 'pull_request' && matrix.mode == 'compile'
+      uses: JamesIves/github-pages-deploy-action@3.7.1
+      with:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        BRANCH: gh-pages
+        FOLDER: /tmp/pages
+        CLEAN: true
+
+
+  Build:
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - { os: 'ubuntu:xenial' }
+          - { os: 'ubuntu:bionic' }
+          - { os: 'ubuntu:eoan' }
+          - { os: 'ubuntu:focal' }
+          - { os: 'ubuntu:groovy' }
+#         - { os: 'centos:6' }
+          - { os: 'centos:7' }
+          - { os: 'centos:8' }
+    name: ${{ matrix.os }}
+    env:
+      DOCKER_JOB: ${{ matrix.os }}
+    steps:
+
+    - uses: actions/checkout@v2
+      with:
+       # Download complete repository + tags
+       fetch-depth: 0
+
+    - run: docker pull ${{ matrix.os }}
+
+    - name: Build container image 'basic'
+      run: source ./.github/settings.sh; ./releasing/docker-run.sh basic
+
+    - name: Build container image 'bazel'
```

Comment:
I don't quite get this step?

Reply:
Depending on the OS version, the 'basic' image is built with one or other last stage in the multistage dockerfile. Then, depending on the OS version too, Bazel needs to be installed using one or other procedure. Therefore, instead of trying to handle the complexity in a single dockerfile, it is easier to have it in two. In practice, image 'basic' is a temporary cache of image 'bazel', that's why one step is executed right after the other.

My end goal is to execute "Build container image 'basic'" and "Build container image 'bazel'" in a different workflow (probably once a week) and publish 'bazel' images only to a registry (tagged as say `google/verible:ubuntu-bionic_build`). Then, the "regular" CI run would be reduced to building Verible in those pre-built container images.